app/src/androidTest/java/org/mozilla/focus/activity/screenshots/ErrorPagesScreenshots.java

```diff
@@ -0,0 +1,99 @@
+import org.mozilla.focus.activity.MainActivity;
+import org.mozilla.focus.activity.TestHelper;
+import org.mozilla.focus.activity.helpers.MainActivityFirstrunTestRule;
+
+import tools.fastlane.screengrab.Screengrab;
+import tools.fastlane.screengrab.locale.LocaleTestRule;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.action.ViewActions.pressImeActionButton;
+import static android.support.test.espresso.action.ViewActions.replaceText;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.hasFocus;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.web.sugar.Web.onWebView;
+import static android.support.test.espresso.web.webdriver.DriverAtoms.findElement;
+import static android.support.test.espresso.web.webdriver.DriverAtoms.webClick;
+import static android.support.test.espresso.web.webdriver.DriverAtoms.webScrollIntoView;
+import static junit.framework.Assert.assertTrue;
+
+@RunWith(AndroidJUnit4.class)
+public class ErrorPagesScreenshots extends ScreenshotTest {
+    @Rule
+    public ActivityTestRule<MainActivity> mActivityTestRule = new MainActivityFirstrunTestRule(false);
+
+    @ClassRule
+    public static final LocaleTestRule localeTestRule = new LocaleTestRule();
+
+    private enum ErrorTypes {
+        ERROR_UNKNOWN (-1),
+        ERROR_HOST_LOOKUP (-2),
+        ERROR_CONNECT (-6),
+        ERROR_TIMEOUT (-8),
+        ERROR_REDIRECT_LOOP (-9),
+        ERROR_UNSUPPORTED_SCHEME (-10),
+        ERROR_FAILED_SSL_HANDSHAKE (-11),
+        ERROR_BAD_URL (-12),
+        ERROR_TOO_MANY_REQUESTS (-15);
+        private int value;
+
+        ErrorTypes(int value) {
+            this.value = value;
+        }
+    }
+
+    @Test
+    public void takeScreenshotsOfErrorPages() throws Exception {
+        for (ErrorTypes error: ErrorTypes.values()) {
+            onView(withId(R.id.url_edit))
+                    .check(matches(isDisplayed()))
+                    .check(matches(hasFocus()))
+                    .perform(click(), replaceText("error:" + error.value), pressImeActionButton());
+
+            assertTrue(TestHelper.webView.waitForExists(waitingTime));
+            assertTrue(TestHelper.progressBar.waitUntilGone(waitingTime));
+
+            // Android O has an issue with using Locator.ID
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                UiObject tryAgainBtn = device.findObject(new UiSelector()
+                        .descriptionContains(getString(R.string.errorpage_refresh))
+                        .clickable(true));
+                assertTrue(tryAgainBtn.waitForExists(waitingTime));
+            } else {
+                onWebView()
```

Comment:
So when I run a suite locally on the sim, there are no issues, but when I run this test alone, I always run into an error here: 

Caused by: java.util.concurrent.TimeoutException: Waited 10 seconds for android.support.test.espresso.core.internal.deps.guava.util.concurrent.AbstractTransformFuture$TransformFuture@2c24f97[status=PENDING, info=[inputFuture=[android.support.test.espresso.core.internal.deps.guava.util.concurrent.SettableFuture@6c61584[status=PENDING]], function=[android.support.test.espresso.web.action.AtomAction$3@4fd916d]]]

This seems like some internal espresso process is blocked by something.  This is also why I replaced this part with the uiautomator PR. (https://github.com/mozilla-mobile/focus-android/issues/1415)

Reply:
On what configuration are you running the tests?